src/blockrelay/compactblock.cpp

```diff
@@ -131,7 +131,7 @@ bool CompactBlock::HandleMessage(CDataStream &vRecv, CNode *pfrom)
     CompactBlock tmp;
     vRecv >> tmp;
     auto pblock = thinrelay.SetBlockToReconstruct(pfrom, tmp.header.GetHash());
-    pblock->cmpctblock = std::make_shared<CompactBlock>(std::forward<CompactBlock>(tmp));
+    pblock->cmpctblock = std::make_shared<CompactBlock>(tmp);
```

Comment:
this will copy tmp into a new shared object.  It would be best to avoid this copy by making the first declaration (of tmp) a shared object.  Then the pblock->cmpctblock assignment just adds to the refcount of the shared pointer.  

Same for the other 3 changes...

Reply:
you're right, after reviewing i think it was fine the way it was.  Closing.